src/debug.cc

```diff
@@ -71,19 +98,359 @@ class DebugFile
     FILE *file_ = nullptr; ///< opened "real" file or nil; never stderr
 };
 
-/// configured cache.log file or stderr
+/// meta-information of a Finish()ed debugs() message
+class DebugMessageHeader
+{
+public:
+    DebugMessageHeader(const DebugRecordCount aRecordNumber, const Debug::Context &);
+
+    DebugRecordCount recordNumber; ///< LogMessage() calls before this message
+    time_t timestamp; ///< approximate debugs() call time
+    int section; ///< debugs() section
+    int level; ///< debugs() level
+    bool forceAlert; ///< debugs() forceAlert flag
+};
+
+// Avoid SBuf for CompiledDebugMessageBody:
+// * SBuf's own debugging may create a lot of reentrant debugging noise.
+// * Debug::Context::buf is an std::string-based STL ostream. Converting its
+//   buf() result to a different kind of string may increase complexity/cost.
+// TODO: Consider switching to a simple fixed-size buffer and a matching stream!
+/// The processed "content" (i.e. the last parameter) part of a debugs() call.
+using CompiledDebugMessageBody = std::string;
+
+/// a fully processed debugs(), ready to be logged
+class CompiledDebugMessage
+{
+public:
+    using Header = DebugMessageHeader;
+    using Body = CompiledDebugMessageBody;
+
```

Comment:
This class is quite simple. No need to complicate it with these typedef equivalents.

Reply:
or, if you have a good reason for adding them. Please add a comment to the class to say why so they do not get polished away in future.

Reply:
Left as is. If you insist, I will remove these `using` declarations, but I believe they make the code a bit more readable/clear and do not increase complexity. They will also minimize the noise in the diff when/if we refactor to get rid of the Debug "class" add a namespace for the debugging-related APIs.

I doubt anybody accidentally removes these declarations. If they do get removed, then somebody would probably have a good reason for that removal, so no problem there.